### ISSUE TITLE
Mesosphere: DC/OS Integration

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#507]
We need this. More details TBD.

**Comment by bdarnell** [comment]
For reference: https://github.com/cockroachdb/cockroach/issues/6859

**Comment by jseldess** [comment]
@a-robinson, in the docs meeting today, we learned that docs for running cockroachdb on mesosphere have become a priority. Do you have experience with mesos? Think you could help me get started?

**Comment by a-robinson** [comment]
I'm familiar with its architecture and high-level features, but not particularly familiar with the details of actually using it.

What sort of help would you like? Generally figuring out the best way to fit Cockroach onto it or something more specific?

**Comment by jseldess** [comment]
@a-robinson, just generally figuring out best way to fit Cockroach into it. We'll want a tutorial just like we have for kubernetes and docker swarm, where we run a multi-node cluster across containers, with stable identity and storage for nodes. Perhaps I should do some preliminary research and the pull you in when I see (or don't see) a way forward?

**Comment by a-robinson** [comment]
Either way works for me. I don't at all mind digging into this stuff. I should have time to start looking into it on Monday.

**Comment by jseldess** [comment]
You'll be a lot quicker than me with this, so if you have time, that's great. Thanks, Alex!

**Comment by bdarnell** [comment]
@a-robinson I got cockroachdb running in mesosphere here: https://github.com/cockroachdb/cockroach/issues/6859#issuecomment-221145692

**Comment by jseldess** [comment]
Putting this on hold until Q1 in favor of docs on k8 and swarm secure deployment.

**Comment by jseldess** [comment]
@a-robinson, once this is ready for docs, please apply the `ready` label and add links to relevant PRs. Thanks!

**Comment by bergerx** [comment]
Has been anything done for this other than #6859?
Are there any plans for developing a Mesos / DC/OS framework for CockroachDB?

**Comment by a-robinson** [comment]
Hi @bergerx, development of a DC/OS framework is mostly complete. It just needs to be published, which should happen soon, likely within the next week or two.

**Comment by jseldess** [comment]
@a-robinson, please let me know what I or someone from docs can do to help here. I'll mark this as `ready`.

**Comment by a-robinson** [comment]
I'll send you all the resources in the meantime while I work on getting them published.

**Comment by bergerx** [comment]
Any update on this?

**Comment by jseldess** [comment]
@a-robinson, if we want mesos orchestration docs for 1.1, any chance you can help with the first draft?

**Comment by laferrieren** [comment]
Highly interested in this, if there is anything that I can do from outside (including reviewing/writing docs or beta testing the framework) I am more then willing to help out, just let me know.

**Comment by bdarnell** [comment]
We just announced a DC/OS package [yesterday at Mesoscon](http://events.linuxfoundation.org/events/mesoscon-north-america/program/schedule). There is some [documentation](https://github.com/dcos/examples/tree/master/cockroachdb/1.9) in the DC/OS examples repo.

**Comment by a-robinson** [comment]
As @bdarnell alluded to, the DC/OS package has been up on Universe for a while now, and the source is now up at https://github.com/cockroachdb/dcos-cockroachdb-service. Please try it out, file issues, and submit any improvements you put together!
